sdk/eventhub/event-hubs/src/sender.ts

```diff
@@ -71,12 +99,32 @@ export class EventHubProducer {
    * @throws {TypeError} Thrown if a required parameter is missing.
    * @throws {Error} Thrown if the underlying connection or sender has been closed.
    * @throws {Error} Thrown if a partitionKey is provided when the producer was created with a partitionId.
+   * @throws {Error} Thrown if batch was created with partitionKey different than the one provided in the options.
    * Create a new producer using the EventHubClient createProducer method.
    */
-  async send(eventData: EventData | EventData[], options?: SendOptions): Promise<void> {
+  async send(
+    eventData: EventData | EventData[] | EventDataBatch,
+    options?: SendOptions
+  ): Promise<void> {
     this._throwIfSenderOrConnectionClosed();
     throwTypeErrorIfParameterMissing(this._context.connectionId, "eventData", eventData);
-    if (!Array.isArray(eventData)) {
+
+    if (eventData instanceof EventDataBatch) {
+      throwTypeErrorIfParameterMissing(this._context.connectionId, "events", eventData.events);
```

Comment:
`events` is always initialized as an empty array in the `EventDataBatch` class, so I don't think this would ever throw.

Reply:
The fact that the `instanceof` check passes means that there is no need for the parameter missing check